website/sales/api/v2/views.py

```diff
@@ -0,0 +1,63 @@
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import filters
+from rest_framework.generics import (
+    ListAPIView,
+    RetrieveAPIView,
+    get_object_or_404,
+    RetrieveUpdateDestroyAPIView,
+    ListCreateAPIView,
+)
+from rest_framework.permissions import DjangoModelPermissionsOrAnonReadOnly, \
+    IsAuthenticated, DjangoModelPermissions
+
+from sales.api.v2.serializers.order import OrderSerializer, OrderListSerializer
+from sales.api.v2.serializers.shift import ShiftSerializer
+from sales.models.order import Order
+from sales.models.shift import Shift
+
+
+class ShiftListView(ListAPIView):
+    """Returns an overview of all sales shifts."""
+
+    serializer_class = ShiftSerializer
+    queryset = Shift.objects.all()
+    filter_backends = (
+        filters.OrderingFilter,
+        filters.SearchFilter,
+    )
+    ordering_fields = ("start_date", "end_date")
+    permission_classes = [IsAuthenticated]
+
+
+class ShiftDetailView(RetrieveAPIView):
+    serializer_class = ShiftSerializer
+    queryset = Shift.objects.all()
+    permission_classes = [IsAuthenticated]
```

Comment:
This needs scope permissions and probably even the DjangoModelPermissions authenticator